lib/max/other/lib-acl.inc.php

```diff
@@ -155,47 +161,76 @@ function MAX_AclSave($acls, $aEntities, $page = false)
         return true; // No changes to the ACL
     }
 
-    $doAcls = OA_Dal::factoryDO($aclsTable);
-    $doAcls->whereAdd($fieldId.' = '.$aclsObjectId);
-    $doAcls->delete(true);
+    if ($mysqlInUse) {
+        // Store the original ACLS table info, in case of data truncation
+        $originalAcls = array();
+        $doAcls = OA_Dal::factoryDO($aclsTable);
+        $doAcls->whereAdd($fieldId.' = '.$aclsObjectId);
+        $doAcls->find();
+        while ($doAcls->fetch()) {
+            $originalAcls[] = $doAcls->toArray();
+        }
+    }
 
-    if (!empty($acls))
-    {
-        foreach ($acls as $index => $acl)
+    // Delete the current master delivery rule data from the DB
+    if (MAX_AclDeleteValues($aclsTable, $fieldId, $aclsObjectId) === false) {
+        $session['aclsDbError'] = true;
+        return;
```

Comment:
The function seems to return a bool, yet a null is returned here and a few times later. Using `return false` is probably better for consistency.

Reply:
Done.